development/db/structuredConditionals.rst

```diff
@@ -0,0 +1,504 @@
+2. Linq -> I didn't know the assistance of Linq until today. From what I searched, not only it has the same issue as Doctrine, while also its interface is unnecessarily complex for the common folk who just wants to change a small amount of information.
+
+
+The Data structure
+=============
+
+This builder uses a tree-like information organization for the boolean comparisons in SQL queries.
+Each node of such tree is a php array.
+Each node can have one of 3 formats:
+
+type1
+-------
+
+The 1st type contains 3 elements:
+
+Left hand, operator, right hand.
+E.g.
+
+.. code-block:: php
+
+	array('f.forum_id', '=', 1)
+	array('f.forum_id', '<>', 1)
+	array('f.forum_id', 'IN', array())
+	array('f.forum_id', 'IN', array(1,2,5,6,7))
+	array('f.forum_id', 'NOT_IN', array(1,2,5,6,7))
+	array('f.forum_id', 'IS', NULL)
+
+For the operator, there are 6 special values (everything else is taken literally):
+
+1. IN
+2. NOT_IN
+3. LIKE
+4. NOT_LIKE
+5. IS
+6. IS_NOT
+
+All of them are special because they call the dbal's methods to process the data.  
+For example, if you use the **IN** operator, it calls $db->sql_in_set() with the right hand data.
+
+type2
+-------
+
+The 2nd type is variable length. It is identified by having the string 'AND', 'OR' or 'NOT' in the first position of the array.
+
+The first element contains the boolean operator that is used to join together all its other elements.
+
+E.g.
+
+.. code-block:: php
+
+		array('OR',
+			array('t.forum_id', '=', 3),
+			array('t.topic_type', '=', 0),
+			array('t.topic_id', 'IN', array(2,3,4)),
+		)
+
+which outputs (after reindenting)
+
+.. code-block:: php
+
+		t.forum_id = 3 OR
+		t.topic_type = 0 OR
+		t.topic_id IN (2, 3, 4)
+
+
+type3
+-------
+
+The 3rd type has 5 elements  
+Left hand, operator, sub query operator, sub query SELECT type, the sub query.
+
+This is used when you require a subquery in your DB query.  
+Essentially, what this does is that it will call sql_build_query() recursively with the 4th and the 5th elements.
+
+.. code-block:: php
+
+	array('f.forum_id', '=', 'ANY', 'SELECT', array(
+						'SELECT' => array(/*...*/),
+						'FROM' => array(/*...*/),
+					)
+	)
+
+	array('f.forum_id', '', 'IN', 'SELECT', array(
+						'SELECT' => array(/*...*/),
+						'FROM' => array(/*...*/),
+					)
+	)
+
+Why arrays?
+=============
+
+The motivation to use arrays comes from the needs:
+
+1. This is information that is going to be used quite a lot.
+	1.1. In the ideal case, every SQL query with either an ON or a WHERE clause (just about all) will use this.
+2. The implementation on which this works on top of already uses arrays.
+3. Editing arrays is a quite trivial task for any piece of code.
+
+Why not Objects?
+-------
+
+1. Tranversing Objects forming a tree is **seriously slow** in php.
```

Comment:
This shows as bold and italic on
https://github.com/brunoais/documentation/blob/feature/structuredConditionals/development/db/structuredConditionals.rst#why-arrays
How do I solve it? Suggestions?